lib/modules/ipfs/upload.js

```diff
@@ -12,48 +12,51 @@ class IPFS {
   }
 
   deploy() {
-    console.log("deploying!");
-    let self = this;
-    async.waterfall([
-      function findBinary(callback) {
-        let ipfs_bin = shelljs.which(self.configIpfsBin);
-
-        if (ipfs_bin === 'ipfs not found' || !ipfs_bin) {
-          console.log(('=== WARNING: ' + self.configIpfsBin + ' not found or not in the path. Guessing ~/go/bin/ipfs for path').yellow);
-          ipfs_bin = "~/go/bin/ipfs";
+    return new Promise((resolve, reject) => {
+      console.log("deploying!");
+      let self = this;
+      async.waterfall([
+        function findBinary(callback) {
+          let ipfs_bin = shelljs.which(self.configIpfsBin);
+
+          if (ipfs_bin === 'ipfs not found' || !ipfs_bin) {
+            console.log(('=== WARNING: ' + self.configIpfsBin + ' not found or not in the path. Guessing ~/go/bin/ipfs for path').yellow);
+            ipfs_bin = "~/go/bin/ipfs";
+          }
+
+          return callback(null, ipfs_bin);
+        },
+        function runCommand(ipfs_bin, callback) {
+          let cmd = `"${ipfs_bin}" add -r ${self.buildDir}`;
+          console.log(("=== adding " + self.buildDir + " to ipfs").green);
+          console.log(cmd.green);
+          let result = shelljs.exec(cmd);
```

Comment:
Since we are in an async waterfall, we should use the async version of exec.

Reply:
Shouldn't this stay as a synchronous `exec` so that the result of the cmd can be passed to the next function in the waterfall?

Reply:
You can get the result in the callback.
https://github.com/shelljs/shelljs/blob/master/README.md#execcommand--options--callback